src/components/kubectl/port-forward.ts

```diff
@@ -90,23 +96,50 @@ export async function portForwardKubernetes (kubectl: Kubectl, explorerNode?: an
             host.showErrorMessage("Error while selecting pod for port-forward");
             return;
         }
-
-        const portMapping = await promptForPort();
         const namespace = await kubectlUtils.currentNamespace(kubectl);
-        portForwardToPod(kubectl, podSelection, portMapping, namespace);
+        const portMapping = await promptForPort(kubectl, podSelection, namespace);
+        if (portMapping.length !== 0) {
+            portForwardToPod(kubectl, podSelection, portMapping, namespace);
+        }
     }
 }
 
 /**
  * Prompts the user on what port to port-forward to, and validates numeric input.
  * @returns An array of PortMapping objects.
  */
-async function promptForPort (): Promise<PortMapping[]> {
+async function promptForPort (kubectl?: Kubectl, podName?: string, namespace?: string): Promise<PortMapping[]> {
     let portString: string | undefined;
+    let defaultValue = undefined;
+    if (podName && kubectl) {
+        const ns = namespace || 'default';
+        try {
+            const result = await kubectl.invokeAsync(`get pods ${podName} --namespace ${ns} -o json`);
+            if (result && result.code === 0) {
+                const pod = JSON.parse(result.stdout) as kubernetes.V1Pod;
+                const containers = pod.spec.containers;
+                const ports = [] as number[];
+                containers.forEach((container: kubernetes.V1Container) => {
```

Comment:
Do you need the type declaration here?  TypeScript is usually pretty good at inferring this (assuming it's inferred `containers` `Container[]` rather than `any`!).

Reply:
removed.